microsoft-edge/web-platform/ad-selection-api.md

```diff
@@ -0,0 +1,297 @@
+| **K-Anonymity** | Collects k-anonymity counters and checks that the winning ad banner passes the k-anonymity check. |
+
+
+<!-- ------------------------------ -->
+#### User-Defined Functions
+
+Sellers and buyers need to provide their own custom code as User-Defined Functions (UDFs) that run in private containers within the deployed services.  These User-Defined Functions can execute custom business logic.
+
+Each buyer or seller can deploy multiple instances of a given service, such as:
+* **SellerFrontEnd** and **Auction**.
+* **BuyerFrontend**, **Bidding**, and **K-Anonymity**.
+
+Some of these services take an input which is a JavaScript worklet that defines User-Defined Functions.
+
+
+<!-- ---------- -->
+###### User-Defined Functions for sellers
+
+| UDF | Service in which the UDF runs | Description |
+| --- | --- | --- |
+| `scoreAd()` | **Auction** | Generates a score for each buyer's bid, or rejects the bid.  This score is then used by the **SellerFrontEnd** service, to choose a winner from among all buyers. |
+| `reportResult()` | **SellerFrontEnd** | For event-level reporting.  Runs in the **SellerFrontEnd** service after the final winner has been chosen.  Notifies the seller about the winning bidder, and provides the bid value. |
+
+
+<!-- ---------- -->
+###### User-Defined Functions for buyers
+
+| UDF | Service in which the UDF runs | Description |
+| --- | --- | --- |
+| `generateBids()` | **Bidding** | Generates a bid, and chooses the banner that will be shown. |
+| `reportWin()` | **SellerFrontEnd** | For event-level reporting.  Runs in the **SellerFrontEnd** service, if the buyer's bid wins the auction.  Notifies the buyer that they have won the auction, and generates notification URLs that will be triggered later, such as during banner rendering or other client events. |
+| `getValues()` | **Bidding Selection & Key/Value** | Runs in the **Bidding Selection & Key/Value** service, if the buyer uses that service.  Allows for custom code execution for data lookup and selection. |
+
+
+<!-- ------------------------------ -->
+#### Deployment guide
+
+To deploy the Ad Selection API service to Azure:
+
+1. Create an Azure account and create a subscription at [Microsoft Azure](https://azure.microsoft.com).
+
+1. Choose the **Confidential ACO containers** option.
+
+1. Download the Terraform [deployment scripts and guide](https://go.microsoft.com/fwlink/?linkid=2290115).  **Note:** Make sure the hash of the image you downloaded matches the hash in the script you download.
```

Comment:
Download the Terraform [deployment scripts and guide](https://go.microsoft.com/fwlink/?linkid=2290115). 

I assume this link need to be changed just before release(we need publish these scripts first)

Reply:
the definition of https://go.microsoft.com/fwlink/?linkid=2290115 will need to be updated.
that url in .md file is ok/stable.

Reply:
closing this, b/c scripts are being published now.
Terraform deployment scripts and guide
https://go.microsoft.com/fwlink/?linkid=2290115

how to view or update the definition of a fwlink:
https://review.learn.microsoft.com/en-us/help/contribute/contribute-link-to-articles-from-the-user-interface?branch=main#create-an-fwlink -- AKA Link Manager tool > set Host to go.microsoft.com
@yanushP